ui/src/app/settings/actions/config.js

```diff
@@ -13,4 +13,18 @@ config.fetch = () => {
   };
 };
 
+config.save = (name, value) => {
```

Comment:
This needs to be an array - dispatching an action for each keypair would mean we make a ws request for every form field.

Reply:
I agree that would be preferable, but it doesn't seem possible. I've tried sending this as an array and the app crashes. Looking at the websocket handler I came to the same conclusion as @Caleb-Ellis in [his comment here](https://github.com/canonical-web-and-design/maas-ui/pull/63#discussion_r311912068).

Reply:
Not just preferable, a requirement, so this branch will need to hold until that is fixed.

Reply:
Support for [params as an array has landed](https://github.com/canonical-web-and-design/maas-ui/commit/d71059c4b9709ae0b50c91c4cb7428322f43fb0e), so this can now be `config.save = params =>`

Reply:
Best to manipulate the array here (so that it has the form `[{name: 'a': value: 'b'}, {...}]`, rather than the calling code.

Reply:
done